eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -505,16 +512,16 @@ object PeerConnection {
   case class PendingAuth(connection: ActorRef, remoteNodeId_opt: Option[PublicKey], address: InetSocketAddress, origin_opt: Option[ActorRef], transport_opt: Option[ActorRef] = None) {
     def outgoing: Boolean = remoteNodeId_opt.isDefined // if this is an outgoing connection, we know the node id in advance
   }
-  case class Authenticated(peerConnection: ActorRef, remoteNodeId: PublicKey)
-  case class InitializeConnection(peer: ActorRef, chainHash: ByteVector32, features: Features, doSync: Boolean)
-  case class ConnectionReady(peerConnection: ActorRef, remoteNodeId: PublicKey, address: InetSocketAddress, outgoing: Boolean, localInit: wire.Init, remoteInit: wire.Init)
+  case class Authenticated(peerConnection: ActorRef, remoteNodeId: PublicKey) extends RemoteTypes
+  case class InitializeConnection(peer: ActorRef, chainHash: ByteVector32, features: Features, doSync: Boolean) extends RemoteTypes
+  case class ConnectionReady(peerConnection: ActorRef, remoteNodeId: PublicKey, address: InetSocketAddress, outgoing: Boolean, localInit: wire.Init, remoteInit: wire.Init) extends RemoteTypes
 
-  sealed trait ConnectionResult
+  sealed trait ConnectionResult extends RemoteTypes
   object ConnectionResult {
     sealed trait Success extends ConnectionResult
     sealed trait Failure extends ConnectionResult
 
-    case class NoAddressFound(remoteNodeId: PublicKey) extends ConnectionResult.Failure { override def toString: String = "no address found" }
+    case object NoAddressFound extends ConnectionResult.Failure { override def toString: String = "no address found" }
```

Comment:
@btcontract I reverted that change made in #1594. Please don't just randomely add a `remoteNodeId` there because you need it, and break consistency among the different `ConnectionResult` types. I'm not opposed to adding it at the `ConnectionResult` level.

Reply:
Actually just tried it and this creates issues because in some cases (e.g. for `ConnectionResult.AuthenticationFailed`) we don't have it. Sigh...